tox.ini

```diff
@@ -10,12 +10,26 @@ deps =
     graphql-core>=2.1
     pytest-cov
 commands =
-    py{py,27,34,35,36}: py.test tests {posargs}
+    py{py,27,34,35,36}: py.test --cov=graphql_server tests {posargs}
+
+[testenv:pre-commit]
+basepython=python3.6
+deps =
+    pre-commit>0.12.0
+setenv =
+    LC_CTYPE=en_US.UTF-8
+    LC_ALL=C.UTF-8
+    LANG=C.UTF-8
+
+
+commands =
+    pre-commit {posargs:run --all-files}
 
 [testenv:flake8]
 basepython=python3.6
 deps = flake8
 commands =
+    pip install -e .
```

Comment:
I don't usually see `pip install` in a tox.ini testenv since tox takes care of creating virtualenv for you with the `deps` specified. I'm not that familiar with this so feel free to ignore this comment, but is there a more tox-y way to do this?

Reply:
If so, can we make the same change on graphene? I was referring to that as the source of truth on how to do it :)
https://github.com/graphql-python/graphene/blob/master/tox.ini